src/app/carros/carros.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnInit } from '@angular/core';
+import {FormControl, FormGroup, Validators} from '@angular/forms';
+import { MatDialog } from '@angular/material/dialog';
+import { Carro } from '../models/Carro';
+
+@Component({
+  selector: 'app-carros',
+  templateUrl: './carros.component.html',
+  styleUrls: ['./carros.component.css']
+})
+export class CarrosComponent implements OnInit {
+
+  carrosForm = new FormGroup ({
+    marca: new FormControl('', Validators.required),
+    nombre: new FormControl('', Validators.required),
+    modelo: new FormControl('', Validators.required),
+    capacidadMotor: new FormControl('', Validators.required),
+    color: new FormControl('', Validators.required)
+  });
+
+  carros: Array <Carro> = [];
+  public mostrar: boolean = false;
+
+  promedio() {
+    let suma = 0;
+    this.carros.forEach(c => {
+      suma += c.capacidadMotor;
+    });
+    if(suma > 0){
```

Comment:
porque se verifica suma? y que pasa si this.carros.length = 0 ???  saca una excepcion no?  corregir.